fuzz/provider.c

```diff
@@ -0,0 +1,687 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * https://www.openssl.org/source/license.html
+ * or in the file LICENSE in the source distribution.
+ */
+#include <string.h>
+#include "openssl/types.h"
+#include "openssl/crypto.h"
+#include "openssl/core_names.h"
+#include "openssl/kdf.h"
+#include "openssl/evp.h"
+#include "openssl/provider.h"
+#include "fuzzer.h"
+
+#define DEFINE_ALGORITHMS(name, evp) DEFINE_STACK_OF(evp) \
+    static int cmp_##evp(const evp *const *a, const evp *const *b); \
+    static void collect_##evp(evp * digest, void * stack); \
+    static void init_##name(OSSL_LIB_CTX * libctx); \
+    static void cleanup_##name(void); \
+    static STACK_OF(evp) * name##_collection; \
+    static int cmp_##evp(const evp *const *a, const evp *const *b) \
+    { \
+        return strcmp(OSSL_PROVIDER_get0_name(evp##_get0_provider(*a)), \
+                      OSSL_PROVIDER_get0_name(evp##_get0_provider(*b))); \
+    } \
+    static void collect_##evp(evp * digest, void * stack) \
```

Comment:
The digest name here seems curious, since this seems to be more than just digests.